trinity/protocol/common/managers.py

```diff
@@ -82,7 +85,12 @@ def _handle_msg(self, msg: TResponse) -> None:
         request, future = self.pending_request
 
         try:
-            request.validate_response(msg)
+            response = await self._normalize_response(msg)
```

Comment:
@gsalgado this is the mechanism that now allows us to do avoid doing heavy computations on response data both during validation and processing.  Whatever *pre-processing* that *must* be done on response data in order to validate it is now required to be done before validation.  Each `Manager` class will then implement the pre-processing step via a `_normalize_response` method, and the data returned is in whatever format is deemed appropriate for the given response type.

Here, for `NodeData` responses, the returned format is a `Tuple[Tuple[Hash32, bytes], ...]`, aka a tuple of `(keccak(node), node)` pairs.  For `Receipts` I think this is going to be a 3-tuple of `[receipts, receipts_root, trie_kv_data]`, ie. the receipts paired up with the return of `mk_trie_nodes_and_data`.  Happy to hear feedback on this as it's the best I could come up with.

Reply:
That sounds good to me!